xline/src/storage/index.rs

```diff
@@ -134,45 +180,53 @@ impl IndexOperate for Index {
         revision: i64,
         sub_revision: i64,
     ) -> Vec<(Revision, Revision)> {
-        let mut index = self.index.lock();
-        match RangeType::get_range_type(key, range_end) {
-            RangeType::OneKey => index
-                .get_mut(key)
-                .and_then(|revs| {
-                    Self::get_revision(revs, 0)
-                        .map(|rev| {
-                            let del_rev = KeyRevision::new_deletion(revision, sub_revision);
-                            revs.push(del_rev);
-                            (rev, del_rev.as_revision())
-                        })
-                        .map(|rev_pair| vec![rev_pair])
-                })
-                .unwrap_or_default(),
-            RangeType::AllKeys => index
-                .values_mut()
-                .zip(0..)
-                .filter_map(|(revs, i)| {
-                    Self::get_revision(revs, 0).map(|rev| {
-                        let del_rev =
-                            KeyRevision::new_deletion(revision, sub_revision.overflow_add(i));
-                        revs.push(del_rev);
-                        (rev, del_rev.as_revision())
+        let mut inner = self.inner.lock();
+
+        let (pairs, keys) = match RangeType::get_range_type(key, range_end) {
+            RangeType::OneKey => {
+                let pairs: Vec<(Revision, Revision)> = inner
+                    .index
+                    .get_mut(key)
+                    .into_iter()
+                    .filter_map(|revs| Self::gen_del_revision(revs, revision, sub_revision))
+                    .collect();
+                let keys = if pairs.is_empty() {
+                    vec![]
+                } else {
+                    vec![key.to_vec()]
+                };
+                (pairs, keys)
+            }
+            RangeType::AllKeys => {
```

Comment:
Same as the `get_from_rev`. Plz enrich the test case `test_delete` to cover the `RangeType::AllKeys` and `RangeType::Range` match arms.

Reply:
Fixed